### ISSUE TITLE
Can't connect to password protected bucket

### ISSUE THREAD

**Issue by tleyden** [couchbaselabs/cbgt#24]
Here's how I'm creating the MainCfg() and specifying the auth params as part of the url

```

    // the connection string, eg: "couchbase:http://user:pass@localhost:8091"
    connect := fmt.Sprintf("couchbase:%v", couchbaseUrl)

    register := "wanted"
    bindHttp := feedParams.BindHttp

    cfg, err := cmd.MainCfg(
        IndexTypeSyncGateway,
        connect,
        bindHttp,
        register,
        feedParams.DataDir,
    )
```

but I'm getting 401 errors when connecting to a bucket that has a password set.  (OTOH if the bucket has no password, it works)

Digging into the CBGT code a bit I found this:

https://github.com/couchbaselabs/cbgt/blob/master/cfg_cb.go#L214-L216

```
func (a *CfgCB) GetCredentials() (string, string, string) {
    return a.bucket, "", a.bucket
}
```

Shouldn't it be returning a non-empty password there?

At that point, `a.url` does contain the password, but it seems to be ignored by GetCredentials()

Also, some logs and a stack trace when `GetCredentials()` is called:

```
2015/08/07 16:29:11 cfgcb.url: http://bucket-1:password@localhost:8091
/Users/tleyden/Development/sync_gateway/src/github.com/couchbaselabs/cbgt/cfg_cb.go:219 (0x14081b)
    (*CfgCB).GetCredentials: debug.PrintStack()
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/pools.go:358 (0x10422a)
    maybeAddAuth: user, pass, _ := ah.GetCredentials()
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/pools.go:469 (0x10555a)
    queryRestAPI: err = maybeAddAuth(req, authHandler)
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/pools.go:494 (0x105c35)
    (*Client).parseURLResponse: return queryRestAPI(c.BaseURL, path, c.ah, out)
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/pools.go:595 (0x106d82)
    ConnectWithAuth: return c, c.parseURLResponse("/pools", &c.Info)
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/cbdatasource/cbdatasource.go:1388 (0x11fa70)
    ConnectBucket: client, err := couchbase.ConnectWithAuth(serverURL, auth)
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/cbdatasource/cbdatasource.go:489 (0x118045)
    (*bucketDataSource).refreshCluster: bucket, err := connectBucket(serverURL, d.poolName, d.bucketName, d.auth)
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/cbdatasource/cbdatasource.go:454 (0x12155a)
    func.001: func() int { return d.refreshCluster() },
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/cbdatasource/cbdatasource.go:1518 (0x1214be)
    ExponentialBackoffLoop: progress := f()
/Users/tleyden/Development/sync_gateway/src/github.com/couchbase/go-couchbase/cbdatasource/cbdatasource.go:455 (0x12162d)
    func.002: sleepInitMS, backoffFactor, sleepMaxMS)
/usr/local/go/src/runtime/asm_amd64.s:2232 (0x40001)
    goexit:
```

**Comment by steveyen** [comment]
Hi Traun,
Nimish ( @nimishzynga ) has a proposed changeset in review on gerrit that's probably related...

```
http://review.couchbase.org/#/c/54227/
```

-- steve

**Comment by steveyen** [comment]
Hi Traun,
Correction, actually after looking more closely at your stack trace & link to code, you are right -- the password shouldn't be empty string.

**Comment by steveyen** [comment]
Fixed with commit a408e67

**Comment by tleyden** [comment]
Thanks!  I will retest today.
